src/Compilers/Core/Portable/Text/CompositeText.cs

```diff
@@ -109,55 +97,36 @@ public override SourceText GetSubText(TextSpan span)
             }
         }
 
-        private void GetIndexAndOffset(int position, out int index, out int offset)
-        {
-            // Binary search to find the chunk that contains the given position.
-            int idx = _segmentOffsets.BinarySearch(position);
-            index = idx >= 0 ? idx : (~idx - 1);
-            offset = position - _segmentOffsets[index];
-        }
-
-        /// <summary>
-        /// Validates the arguments passed to <see cref="CopyTo"/> against the published contract.
-        /// </summary>
-        /// <returns>True if should bother to proceed with copying.</returns>
-        private bool CheckCopyToArguments(int sourceIndex, char[] destination, int destinationIndex, int count)
```

Comment:
I moved `CheckCopyToArguments` to `SourceText` so that other derived classes can use it, and renamed to ValidateCopyToArguments`.